### ISSUE TITLE
SEARCH: Incorrect results when using `caseSensitive: true` config option

### ISSUE THREAD

**Issue by thilgen** [handsontable/hyperformula#1225]
### Description

While adding additional codecov for `SEARCH` the following issues were found when `Hyperformula` is using the `caseSensitive=false` config option
1. The `caseSensitive` option is ignored and always uses a case-insensitive `SEARCH`
2. An incorrect index is being calculated when the pattern is NOT found

These issues can be observed with the following unit test
```
it('bug', () => {
  const engineCaseInsensistive = HyperFormula.buildFromArray(
    [
      ['=SEARCH(".B", "fooBarBaz")'],
      ['=SEARCH(".B", "fooBarBaz", 5)'],
      ['=SEARCH(".b", "fooBarBaz")'],
      ['=SEARCH(".b", "fooBarBaz", 5)'],
    ],
    {useRegularExpressions: true, caseSensitive: false}
  )
  expect(engineCaseInsensistive.getCellValue(adr('A1'))).toEqual(3)
  expect(engineCaseInsensistive.getCellValue(adr('A2'))).toEqual(6)
  expect(engineCaseInsensistive.getCellValue(adr('A3'))).toEqual(3)
  expect(engineCaseInsensistive.getCellValue(adr('A4'))).toEqual(6)

  const engineCaseSensistive = HyperFormula.buildFromArray(
    [
      ['=SEARCH(".B", "fooBarBaz")'],
      ['=SEARCH(".B", "fooBarBaz", 5)'],
      ['=SEARCH(".b", "fooBarBaz")'],
      ['=SEARCH(".b", "fooBarBaz", 5)'],
    ],
    {useRegularExpressions: true, caseSensitive: true}
  )
  expect(engineCaseSensistive.getCellValue(adr('A1'))).toEqual(3) // getting ErrorMessage.PatternNotFound
  expect(engineCaseSensistive.getCellValue(adr('A2'))).toEqual(6) // getting 4
  expect(engineCaseSensistive.getCellValue(adr('A3'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.PatternNotFound)) // getting 3
  expect(engineCaseSensistive.getCellValue(adr('A4'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.PatternNotFound)) // getting 6
  })
```
Or in this JSFiddle example - [hyperformula-bug-search](https://jsfiddle.net/thilgen/nhat8d9w/35/)

### Related issues:
1. https://github.com/handsontable/hyperformula/issues/1224
2. https://github.com/handsontable/hyperformula/pull/1226

### Your environment
Mac + Chrome
Handsontable: v12.3.3 (28/03/2023 12:24:28)
HyperFormula: 2.3.0"

**Comment by thilgen** [comment]
[[PR with Bugfix and Increased Cover](https://github.com/handsontable/hyperformula/pull/1224)](https://github.com/handsontable/hyperformula/pull/1224)

**Comment by thilgen** [comment]
Okay. So that means the current behavior for the second two results in the
example I provided are expected - But the first two are still incorrect?

[image: image.png]

The SEARCH impl in HF makes use of an underlying function that DOES look at
the engine env variables for case-sensitivity - so this might need to be
unwound a bit.

**Comment by thilgen** [comment]
![image](https://user-images.githubusercontent.com/9309340/229524711-10e118eb-4e9a-4e9f-b439-ca4020c21af5.png)

**Comment by AMBudnik** [comment]
Hi @thilgen 

Just a quick note. This issue is also solved within [v2.5.0](https://hyperformula.handsontable.com/guide/release-notes.html#_2-5-0).
